packages/code-du-travail-modeles/src/modeles/conventions/1483_habillement_textiles_commerce_de_detail/salary.ts

```diff
@@ -14,6 +14,26 @@ export type CC1483ReferenceSalaryProps = {
 export class ReferenceSalary1483
   implements IReferenceSalary<SupportedCcIndemniteLicenciement.IDCC1483>
 {
+  mapSituation(
+    args: Record<string, string | undefined>
+  ): ReferenceSalaryProps<SupportedCcIndemniteLicenciement.IDCC1483> {
+    return {
+      salaires: args.salaryPeriods
+        ? (JSON.parse(args.salaryPeriods) as SalaryPeriods[])
+        : [],
+      salairesPendantPreavis: args.salairesPendantPreavis
+        ? JSON.parse(args.salairesPendantPreavis)
+        : undefined,
+    };
+  }
+
+  removeSpecificSituation(
+    args: Record<string, string | undefined>
+  ): Record<string, string | undefined> {
+    delete args.salairesPendantPreavis;
+    return args;
+  }
```

Comment:
Pourquoi on a besoin de faire ça ?

Reply:
+1

Reply:
oui sinon publicode nous crache dessus

Reply:
Tu retires les données que publicodes ne gère pas ? Pourquoi ne pas destructurer ici ? 

```suggestion
  removeSpecificSituationFields(
    {salairesPendantPreavis, ...args}: Record<string, string | undefined>
  ): Record<string, string | undefined> {
    return args;
  }
```

Reply:
On aurait pu utiliser le mapSituation pour identifier les champs ajouté et les retirer ensuite. On aurait pu aussi insérer les champs propre à CC dans un sous ensemble spécifique. Ca peut être des axes d'amélioration pour la suite.